### PR TITLE
Add bottom type

### PR DESCRIPTION
**Pull request by YaphetsH**
I have meet some covert soot type errors in Converter class ,eg. soot.jimple.toolkits.typing.fast.BottomType can't convert successful when analysis some android jar file.
So I fix this issue by my fixed commit,If you have time, merge this fix,please.Thx~

### CODE REVIEW

**Comment by zhangt2333** [comment]
Oh, thank you for the PR! Your input on this sentiment is valuable and one we hadn't considered. Could you provide some testing for it?

**Comment by YaphetsH** [comment]
It's a develop of my android branch which based on tai-e master.
Here is the test jar(framework.jar),which entryPoint is custom by myself

[framework.jar.zip](https://github.com/pascal-lab/Tai-e/files/10931895/framework.jar.zip)
 And the question is in the method <android.widget.RemoteViews$BaseReflectionAction Action initActionAsync(ViewTree viewTree, ViewGroup viewGroup, InteractionHandler interactionHandler, ColorResources colorResources)>,which jimple is to the next code:
```
public final android.widget.RemoteViews$Action initActionAsync(android.widget.RemoteViews$ViewTree, android.view.ViewGroup, android.widget.RemoteViews$InteractionHandler, android.widget.RemoteViews$ColorResources)
    {
        java.lang.Throwable $r8;
        java.lang.invoke.MethodHandle $r5;
        bottom_type r6;
        android.widget.RemoteViews$ViewTree r1;
        int i0, i1, i2;
        android.view.View $r2;
        java.lang.String $r4, r13;
        android.widget.RemoteViews r12;
        android.widget.RemoteViews$ColorResources r11;
        android.widget.RemoteViews$InteractionHandler r10;
        android.widget.RemoteViews$ActionException r14, r15;
        android.widget.RemoteViews$BaseReflectionAction r0;
        android.view.ViewGroup r9;
        java.lang.Class $r3;
        java.lang.StringBuilder r16;
        android.widget.RemoteViews$Action $r7;
        r0 := @this: android.widget.RemoteViews$BaseReflectionAction;
        r1 := @parameter0: android.widget.RemoteViews$ViewTree;
       .....
     label2:
        $r5 = staticinvoke <android.widget.RemoteViews: java.lang.invoke.MethodHandle access$1000(android.widget.RemoteViews,android.view.View,java.lang.String,java.lang.Class,boolean)>(r12, $r2, r13, $r3, 1);
     label3:
        if $r5 == null goto label4;
        return r6;
       ...
    }
 ```
The register r6 is bottom type ,so ur...wrong
So If you want to test this lucky case, you should change the way of soot parse jar in Tai-e with android model,Because there is sth. difference in the android mode in soot;And these err occurs when analyze plan is dead-code detection.

**Comment by zhangt2333** [comment]
I have completed this PR, @silverbullettt could you please review it?

File: src/main/java/pascal/taie/language/type/BottomType.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with Tai-e. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package pascal.taie.language.type;
+
+/**
+ * The bottom type is a subtype of every other type.
```
Comment (silverbullettt) [inline src/main/java/pascal/taie/language/type/BottomType.java:26]: Bottom type represents "no type", not subtype. It is temporary type to adapt Soot front end.

**Comment by zhangt2333** [comment]
@YaphetsH 
Thank you for making your first contribution to Tai-e.

**Pull request merged**